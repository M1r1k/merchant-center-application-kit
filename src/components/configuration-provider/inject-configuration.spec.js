import React from 'react';
import { mount } from 'enzyme';
import injectConfiguration from './inject-configuration';

const TestComponent = () => <div>{'Test'}</div>;
TestComponent.displayName = 'TestComponent';
TestComponent.propTypes = {};

describe('HoC', () => {
  const context = {
    configuration: {
      foo: { config: 'value' },
      bar: { more: 'values' },
      baz: { much: 'amaze' },
    },
  };
  let Enhanced;
  let wrapper;

  describe('when requesting a configuration', () => {
    describe('when the configuration exists', () => {
      beforeEach(() => {
        Enhanced = injectConfiguration('foo')(TestComponent);

        wrapper = mount(<Enhanced />, { context });
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should inject the configuration as a `prop`', () => {
        expect(wrapper.find(TestComponent)).toHaveProp(
          'configuration',
          context.configuration.foo
        );
      });
    });

    describe('when the configuration does not exist', () => {
      beforeEach(() => {
        Enhanced = injectConfiguration('nope')(TestComponent);

        wrapper = mount(<Enhanced />, { context });
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should not inject the configuration as a `prop`', () => {
        expect(wrapper.prop('configuration')).not.toBeDefined();
      });
    });

    describe('when a `propName` is passed', () => {
      const propName = 'someFooProp';

      beforeEach(() => {
        Enhanced = injectConfiguration('foo', propName)(TestComponent);

        wrapper = mount(<Enhanced />, { context });
      });

      it('should output correct tree', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should inject the configuration as a `prop`', () => {
        expect(wrapper.find(TestComponent)).toHaveProp(
          propName,
          context.configuration.foo
        );
      });
    });
  });

  describe('when requesting a nested configuration property', () => {
    beforeEach(() => {
      Enhanced = injectConfiguration(['foo', 'config'])(TestComponent);

      wrapper = mount(<Enhanced />, { context });
    });

    it('should inject the configuration as a `prop`', () => {
      expect(wrapper.find(TestComponent)).toHaveProp(
        'configuration',
        context.configuration.foo.config
      );
    });
  });
});
