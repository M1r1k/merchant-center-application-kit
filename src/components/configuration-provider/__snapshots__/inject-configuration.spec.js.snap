// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoC when requesting a configuration when a \`propName\` is passed should output correct tree 1`] = `undefined`;

exports[`HoC when requesting a configuration when the configuration does not exist should output correct tree 1`] = `undefined`;

exports[`HoC when requesting a configuration when the configuration exists should output correct tree 1`] = `undefined`;
